src/core/components/response-body.jsx

```diff
@@ -59,6 +59,7 @@ export default class ResponseBody extends React.PureComponent {
     url = url || ""
 
     if (
+      content.size > 0 &&
```

Comment:
When this `if` statement executes and `content.size > 0` expression is the first predicate, it is not guaranteed that the `content` is of type `Blob`. In order to guarantee that the `content` is of type `Blobl` we have to change the order of the predicates in the following way:

```js
if (
  (/^application\/octet-stream/i.test(contentType) ||
    (headers["Content-Disposition"] &&
      /attachment/i.test(headers["Content-Disposition"])) ||
    (headers["content-disposition"] &&
      /attachment/i.test(headers["content-disposition"])) ||
    (headers["Content-Description"] &&
      /File Transfer/i.test(headers["Content-Description"])) ||
    (headers["content-description"] &&
      /File Transfer/i.test(headers["content-description"]))) &&
  content.size > 0
) {
```

Reply:
Let's put a couple of tests in at the end of **test/unit/components/response-body.jsx**:

```jsx
  it("should render Download file link for non-empty response", function () {
    props.contentType = "application/octet-stream"
    props.content = new Blob(['"test"'], { type: props.contentType });
    const wrapper = shallow(<ResponseBody {...props} />)
    expect(wrapper.text()).toMatch(/Download file/)
  })

  it("should not render Download file link for empty response", function () {
    props.contentType = "application/octet-stream"
    props.content = new Blob()
    const wrapper = shallow(<ResponseBody {...props} />)
    expect(wrapper.text()).not.toMatch(/Download file/)
  })
```

Reply:
In order for these tests to work you need to modify `setUpDomEnvironment` function in **test/unit/setup.js** and add support for creating URL objects:

```js
  // https://github.com/jsdom/jsdom/issues/1721
  if (typeof global.window.URL.createObjectURL === 'undefined') {
    Object.defineProperty(global.window.URL, 'createObjectURL', { value: () => 'data:text/plain;base64,SGVsbG8sIFdvcmxkIQ==' })
  }
```

Reply:
Thanks @char0n ! 
Yep, the predicate order makes sense. I ran into some issues with unit tests so I didn't make any but the change in `setUpDomEnvironment` helps. Thanks!

I've pushed the requested changes.

Reply:
@kowalczyk-krzysztof thanks for the quick changes.